tests/datasets/test_geo.py

```diff
@@ -334,6 +343,132 @@ def test_no_all_bands(self) -> None:
+        file = str(files_found[0])
+        assert file.endswith(filename)
+
+    @pytest.mark.parametrize(
+        'temp_archive',
+        [
+            os.path.join(
+                'tests',
+                'data',
+                'sentinel2',
+                'S2A_MSIL2A_20220414T110751_N0400_R108_T26EMU_20220414T165533.SAFE',
+            )
+        ],
+        indirect=True,
+    )
+    def test_zipped_specific_file_dir(self, temp_archive: tuple[str, str]) -> None:
+        dir_not_zipped, dir_zipped = temp_archive
+
+        # Listing, instead of hardcoding filepath
+        filename_glob = Sentinel2.filename_glob.format('B02')
+        specific_file_not_zipped = list_directory_recursive(
+            dir_not_zipped, filename_glob
+        )[0]
+        filepath_within_dir = specific_file_not_zipped.replace(dir_not_zipped, '')
+
+        specific_file_zipped = f'zip://{dir_zipped}!{filepath_within_dir}'
+
+        files_found = CustomGeoDataset(paths=specific_file_zipped).files
+        assert len(files_found) == 1
+        file = str(files_found[0])
+        assert file.endswith(filepath_within_dir)
+
+    @pytest.mark.parametrize(
+        'temp_archive',
+        [
+            os.path.join(
+                'tests',
+                'data',
+                'sentinel2',
+                'S2A_MSIL2A_20220414T110751_N0400_R108_T26EMU_20220414T165533.SAFE',
+            )
+        ],
+        indirect=True,
+    )
+    def test_zipped_path_non_existing_file(self, temp_archive: tuple[str, str]) -> None:
+        dir_not_zipped, dir_zipped = temp_archive
+        file_zipped_not_existing = os.path.join(dir_zipped, 'non_existing_file.tif')
+        with pytest.warns(UserWarning, match='Path was ignored.'):
+            assert (
+                len(CustomGeoDataset(paths=f'zip://{file_zipped_not_existing}').files)
+                == 0
+            )
+
+    @pytest.mark.parametrize(
+        'temp_archive',
+        [
+            os.path.join(
+                'tests',
+                'data',
+                'sentinel2',
+                'S2A_MSIL2A_20220414T110751_N0400_R108_T26EMU_20220414T165533.SAFE',
+            )
+        ],
+        indirect=True,
+    )
+    def test_zipped_directory(self, temp_archive: tuple[str, str]) -> None:
+        dir_not_zipped, dir_zipped = temp_archive
+        bands = Sentinel2.rgb_bands
+        transforms = nn.Identity()
+        cache = False
+
+        files_not_zipped = Sentinel2(
+            paths=dir_not_zipped, bands=bands, transforms=transforms, cache=cache
+        ).files
+
+        files_zipped = Sentinel2(
+            paths=f'zip://{dir_zipped}', bands=bands, transforms=transforms, cache=cache
+        ).files
+
+        basenames_not_zipped = [Path(path).stem for path in files_not_zipped]
+        basenames_zipped = [Path(path).stem for path in files_zipped]
+        assert basenames_zipped == basenames_not_zipped
+
+    def test_http_files(self) -> None:
```

Comment:
@sfalkena I added this test, do you think it is sufficient? I found adding more files would make the test significantly slower, so I opted for two.